Several fixes and additions to bash script

vitorgalvao opened the pull request:
`$PWD` doesn’t really work as intended at the top, because it’ll output the directory you’re in, not the script’s directory. You should be able to be in whatever directory you want when running the script.

The second commit are generic bash fixes for consistency and correctness.

wong2 commented on asciinema2gif:
```diff
@@ -5,18 +5,18 @@
 set -e
 
-rundir=$PWD
+readonly rundir="$(basename "${0}")"
```
`rundir` should be an absolute path

vitorgalvao replied:
Done.

vitorgalvao replied:
Done (there are shorter ways, with `realpath`, but those don’t work out-of-the-box with OS X).

wong2 replied:
@vitorgalvao hmm, still not working for me (OS X): 

```
➜  ~/codes/js/asciinema2gifgit:(vitorgalvao-generic-bash-fixes) ✗ ./asciinema2gif
Creating and changing to temporary dir: /var/folders/6b/6nfn6_l124jblt0rmkls8l7w0000gn/T/asciinema2gif.XXX.Fn71yrPe
ERROR: Couldn't open .
ERROR: Cannot open file:///private/var/folders/6b/6nfn6_l124jblt0rmkls8l7w0000gn/T/asciinema2gif.XXX.Fn71yrPe: Path is a directory
```

wong2 replied:
it turns out:
`phantomjs render.js $1` would work
while 
`phantomjs render.js "${1}"` would not

vitorgalvao replied:
That’s because you’re trying to run it with no arguments, which shouldn’t happen, anyway. Was saving that fix for another PR (for logical separation), but [included it now](https://github.com/vitorgalvao/asciinema2gif/commit/66e7dfc897946590aa78e653244685f0e9bee9bc).

vitorgalvao commented on asciinema2gif:
```diff
@@ -5,18 +5,25 @@
 set -e
 
-rundir=$PWD
+readonly program="$(basename "${0}")"
```
Useful in case you (or the user) ever change the script’s name.

vitorgalvao commented on asciinema2gif:
```diff
@@ -5,18 +5,25 @@
+readonly rundir="$(cd "$(dirname "${0}")" && pwd -P)"
+
+# Show instructions when using wrong arguments
+if [[ -z "${1}" ]] || ! [[ "${1}" = 'https://asciinema.org/a/'* ]]; then
```
Will fire when given with no arguments, or when the argument isn’t an asciinema url.

wong2 replied:
great!

vitorgalvao commented:
Went ahead and introduced a check for dependencies in this PR as well (better to check for them beforehand and warn the user than letting the script run all the way and then tell the user it failed because they didn’t have something installed) and changed the issue’s title (since the previous one no longer applied).

vitorgalvao commented on asciinema2gif:
```diff
@@ -5,18 +5,45 @@
+  done
+
+  if [[ "${#missing_deps[@]}" -gt 0 ]]; then
+    tput setaf 1
```
Sets text to red.

vitorgalvao commented on asciinema2gif:
```diff
@@ -5,18 +5,45 @@
+    tput setaf 1
+    echo -e '\nThis script has unmet dependencies. You need to install these first:'
+    printf '  %s\n' "${missing_deps[@]}"
+    tput sgr0
```
Resets text color.

wong2 commented on asciinema2gif:
```diff
@@ -5,18 +5,47 @@
+depends_on convert gifsicle phantomjs
+
+# Show instructions when using wrong arguments
+if [[ -z "${1}" ]] || ! [[ "${1}" = 'https://asciinema.org/a/'* ]]; then
```
should be `/api/`, from README:

> $ ./asciinema2gif https://asciinema.org/api/asciicasts/8332

vitorgalvao replied:
Ah, so this *needs* to be used with the `api` url, correct? I was taking a url from the website. Fixed and clarified the message.

vitorgalvao commented:
Just submitted #12, that includes all the commits from this and more, to make the script work in a different (and I believe better for the user) way.
